disk_store_test.go

```diff
@@ -82,23 +88,26 @@ func TestDiskStore_Delete(t *testing.T) {
 	for key, val := range tests {
 		store.Set(key, val)
 	}
-	for key, _ := range tests {
-		store.Set(key, "")
+
+	// only for tests
+	deletedKeys := []string{"hamlet", "dune", "othello"}
+	//delete few keys
+	for _, k := range deletedKeys {
+		store.Delete(k)
 	}
-	store.Set("end", "yes")
 	store.Close()
 
 	store, err = NewDiskStore("test.db")
 	if err != nil {
 		t.Fatalf("failed to create disk store: %v", err)
 	}
-	for key := range tests {
-		if store.Get(key) != "" {
-			t.Errorf("Get() = %v, want '' (empty)", store.Get(key))
+
+	//check for deletion
+	for _, dkeys := range deletedKeys {
+		actualVal, err := store.Get(dkeys)
+		if actualVal != "" && errors.Is(err, ErrKeyNotFound) {
```

Comment:
⚠️ don't you want the condition to be or? 

```
if actualVal != "" || errors.Is(err, ErrKeyNotFound)
```

Better would be splitting this into two conditions, so that we can log the actual error condition: 

```
if actualVal != "" {}

if !errors.Is(err, ErrKeyNotFound) {}
```

Reply:
I added `&` cause Im checking only for the keys in `deletedKeys` list explicitly. But I get your point, making the change.